src/handlers/web3.ts

```diff
@@ -833,3 +836,64 @@ export const estimateGasLimit = async (
     return estimateGas(estimateGasData, provider);
   }
 };
+
+const HAS_MERGED_DEFAULTS: { [key: string]: boolean } = {
+  [Network.mainnet]: false,
+  [Network.goerli]: false,
+  [Network.ropsten]: false,
+  [Network.rinkeby]: false,
+  [Network.kovan]: false,
+};
+const hasMergedStorage = new MMKV();
+export const getHasMerged = (network: Network) => {
+  const storage = hasMergedStorage.getString(STORAGE_IDS.HAS_MERGED);
+  if (storage) {
+    const data = JSON.parse(storage);
+    logger.debug('GET HAS MERGED DATA: ', data);
+    const hasMerged = data[network];
+    if (hasMerged === undefined) {
+      data[network] = HAS_MERGED_DEFAULTS[network];
+      hasMergedStorage.set(STORAGE_IDS.HAS_MERGED, JSON.stringify(data));
+    }
+    return data[network];
+  } else {
+    hasMergedStorage.set(
+      STORAGE_IDS.HAS_MERGED,
+      JSON.stringify(HAS_MERGED_DEFAULTS)
+    );
+  }
+};
+const getSetMergeStorageKey = (setting: boolean) => (network: Network) => {
+  const storage = hasMergedStorage.getString(STORAGE_IDS.HAS_MERGED);
+  if (storage) {
+    const data = JSON.parse(storage);
+    data[network] = setting;
+    hasMergedStorage.set(STORAGE_IDS.HAS_MERGED, JSON.stringify(data));
+    logger.debug('NEW DATA IN STORAGE: ', data);
+  }
+};
+export const setHasMerged = (network: Network) => {
+  getSetMergeStorageKey(true)(network);
+};
+export const setHasNotMerged = (network: Network) => {
+  getSetMergeStorageKey(false)(network);
+};
+export const checkForTheMerge = async (
+  provider: JsonRpcProvider,
+  network: Network
+) => {
+  const currentHasMerged = getHasMerged(network);
+  logger.debug('CURRENT HAS MERGED: ', currentHasMerged);
+  if (currentHasMerged !== true) {
+    const block = await provider.getBlock('latest');
+    logger.debug('LATEST BLOCK: ', block);
```

Comment:
this one is a bit noisy when starting the app on start, did you want to see an specific param of the block?